client.cc

```diff
@@ -35,178 +37,81 @@ using std::vector;
-using supplyfinder::InventoryInfo;
+using namespace supplyfinder;
 
-class VendorClient {
+class FinderClient {
   public:
-  VendorClient(std::shared_ptr<Channel> vendor_channel):
-  vendor_stub_(Vendor::NewStub(vendor_channel)) {}
-  
-  std::string SayHelloToVendor(const std::string& user) {
-    // Data we are sending to the server.
-    HelloRequest request;
-    request.set_name(user);
-
-    // Container for the data we expect from the server.
-    HelloReply reply;
-
-    // Context for the client. It could be used to convey extra information to
-    // the server and/or tweak certain RPC behaviors.
-    ClientContext context;
-
-    // The actual RPC.
-    Status status = vendor_stub_->SayHello(&context, request, &reply);
-
-    // Act upon its status.
-    if (status.ok()) {
-      return reply.message();
-    } else {
-      std::cout << status.error_code() << ": " << status.error_message()
-                << std::endl;
-      return "RPC failed";
-    }
-  }
+  FinderClient(std::shared_ptr<Channel> channel):
+    stub_(Finder::NewStub(channel)) {}
   
-  InventoryInfo InquireInventoryInfo (uint32_t food_id) {
-    FoodID request;
+  vector<ShopInfo> InquireFoodInfo (uint32_t food_id, uint32_t quantity) {
+    Request request;
     request.set_food_id(food_id);
+    request.set_quantity(quantity);
     
-    InventoryInfo info;
+    ShopInfo reply;
+    vector<ShopInfo> info;
     
     ClientContext context;
     
-    Status status = vendor_stub_->CheckInventory(&context, request, &info);
-    
-    if (status.ok()) {
-      return info;
-    } else {
-      std::cout << status.error_code() << ": " << status.error_message()
-                << std::endl;
-      info.set_price(-1);
-      return info;
-    }
-  }
-  private:
-  std::unique_ptr<Vendor::Stub> vendor_stub_;
-};
-
-class SupplierClient {
- public:
-  SupplierClient(std::shared_ptr<Channel> supplier_channel)
-      : supplier_stub_(Supplier::NewStub(supplier_channel)) {}
-
-  // Assembles the client's payload, sends it and presents the response back
-  // from the server.
-  std::string SayHelloToSupplier(const std::string& user) {
-    // Data we are sending to the server.
-    HelloRequest request;
-    request.set_name(user);
-
-    // Container for the data we expect from the server.
-    HelloReply reply;
+    std::unique_ptr<ClientReader<ShopInfo>> reader (stub_ -> CheckFood(&context, request));
 
-    // Context for the client. It could be used to convey extra information to
-    // the server and/or tweak certain RPC behaviors.
-    ClientContext context;
-
-    // The actual RPC.
-    Status status = supplier_stub_->SayHello(&context, request, &reply);
-
-    // Act upon its status.
-    if (status.ok()) {
-      return reply.message();
-    } else {
-      std::cout << status.error_code() << ": " << status.error_message()
-                << std::endl;
-      return "RPC failed";
+    while (reader->Read(&reply)) {
+        info.push_back(reply);
```

Comment:
Since you are streaming, maybe you can print results as they come in instead of collecting them all and then printing?